tests/test_ranges.py

```diff
@@ -1144,6 +1144,16 @@ def test_union_with_half_finite_range(self):
                 )
             )
 
+        @pytest.mark.xfail
```

Comment:
xfail here since in this commit the test will fail - the xfail is removed later in the PR.